### PR TITLE
[ODS-5193] Record Ownership Phase 2 - Add script to cleanup old tables

### PR DESCRIPTION
**Pull request by tbmeador-dlp**

### CODE REVIEW

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentsVerifier.cs [hunk 1]
```diff
@@ -79,8 +81,65 @@ public class AuthorizationSegmentsVerifier : IAuthorizationSegmentsVerifier
...
+                        claimEndpointValuesForDisplayText.Add("...");
+                    }
+                    
+                    string claimEndpointValuesText = string.Join(", ", claimEndpointValuesForDisplayText);
+                    
+                    if (subjectEndpointNames.Length == 1)
+                    {
+                        return $"Authorization denied. No relationships have been established between the caller's education "
+                            + $"organization id {claimOrClaims} ({claimValueOrValues} [{claimEndpointValuesText}] of {typeOrTypes} {claimEndpointNamesText}) and the requested resource's "
+                            + $"{subjectEndpointNamesText} value. (Authorization Strategy: {authorizationStrategyName})";
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentsVerifier.cs]: fixed it

File: Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentsVerifier.cs [hunk 2]
```diff
@@ -79,8 +81,65 @@ public class AuthorizationSegmentsVerifier : IAuthorizationSegmentsVerifier
...
+                    }
+
+                    return $"Authorization denied. No relationships have been established between the caller's education "
+                        + $"organization id {claimOrClaims} ({claimValueOrValues} [{claimEndpointValuesText}] of {typeOrTypes} {claimEndpointNamesText}) and one of the following properties of "
+                        + $"the requested resource: {subjectEndpointNamesText}. (Authorization Strategy: {authorizationStrategyName})";
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentsVerifier.cs]: fixed it

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/AuthorizationSegmentsToFiltersConverter.cs [hunk 3]
```diff
@@ -50,6 +50,7 @@ public IReadOnlyList<AuthorizationFilterDetails> Convert(IReadOnlyList<ClaimsAut
                             SubjectEndpointName = subjectEndpointName,
                             ClaimEndpointName = claimEndpointName,
                             ClaimValues = g.Select(x => x.Value).Distinct().ToArray(),
+                            AuthorizationStrategyName = segment.AuthorizationStrategyName
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/AuthorizationSegmentsToFiltersConverter.cs]: fixed it

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs [hunk 4]
```diff
@@ -155,7 +155,8 @@ InlineAuthorizationResults PerformInlineClaimsAuthorizations()
                         subsequentAuthorizationSegments.Add(new ClaimsAuthorizationSegment(
                             nonInlinableClaimsEndpoints.ToArray(),
                             claimsAuthorizationSegment.SubjectEndpoint,
-                            claimsAuthorizationSegment.AuthorizationPathModifier));
+                            claimsAuthorizationSegment.AuthorizationPathModifier,
+                            claimsAuthorizationSegment.AuthorizationStrategyName));
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs]: fixed it

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs [hunk 5]
```diff
@@ -208,6 +209,11 @@ private class InlineAuthorizationResults
 
             var authorizationSegments = GetAuthorizationSegments(relevantClaims, authorizationContextPropertyNames, null);
 
+            foreach (var authorizationSegment in authorizationSegments)
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs]: fixed it

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs [hunk 6]
```diff
@@ -222,7 +228,14 @@ private class InlineAuthorizationResults
             BuildAuthorizationSegments(builder, signatureProperties);
 
             // Get the rules for execution
-            return builder.GetSegments();
+            var segments = builder.GetSegments();
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs]: fixed it

File: Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs [hunk 7]
```diff
@@ -73,6 +75,11 @@ public ClaimsAuthorizationSegment(IEnumerable<Tuple<string, object>> claimNamesA
         /// <remarks>By convention, this value will be used as a suffix on the database view used for authorization, enabling alternative paths for authorization.</remarks>
         public string AuthorizationPathModifier { get; }
 
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs]: fixed it

File: Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs [hunk 8]
```diff
@@ -50,11 +50,13 @@ public ClaimsAuthorizationSegment(IEnumerable<Tuple<string, object>> claimNamesA
         public ClaimsAuthorizationSegment(
             IReadOnlyList<AuthorizationSegmentEndpointWithValue> claimsEndpoints,
             AuthorizationSegmentEndpoint subjectEndpoint,
-            string authorizationPathModifier)
+            string authorizationPathModifier,
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs]: fixed it

File: Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs [hunk 9]
```diff
@@ -50,11 +50,13 @@ public ClaimsAuthorizationSegment(IEnumerable<Tuple<string, object>> claimNamesA
...
+            string authorizationStrategyName)
         {
             ClaimsEndpoints = claimsEndpoints;
             SubjectEndpoint = subjectEndpoint;
             AuthorizationPathModifier = authorizationPathModifier;
+            AuthorizationStrategyName = authorizationStrategyName;
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Common/Security/Authorization/ClaimsAuthorizationSegment.cs]: fixed it

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json [hunk 10]
```diff
@@ -908,7 +908,7 @@
 											"",
 											"const responseItems = pm.response.json();",
 											"pm.test(\"Should match with Error Message Response \", () => {    ",
-											"        pm.expect(responseItems.message).to.equal(\"Authorization denied.  The claim does not have any established relationships with the requested resource.\");",
+											"        pm.expect(responseItems.message).to.equal(\"Authorization denied. No relationships have been established between the caller's education organization id claim (value [5001] of type 'CommunityProviderId') and one of the following properties of the requested resource: 'SchoolId', 'StudentUniqueId'. (Authorization Strategy: RelationshipsWithEdOrgsAndPeopleAuthorizationStrategy`1)\");",
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json]: fixed it

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json [hunk 11]
```diff
@@ -1440,7 +1440,7 @@
 											"});",
 											"const responseItems = pm.response.json();",
 											"pm.test(\"Should match with Error Message Response \", () => {    ",
-											"        pm.expect(responseItems.message).to.equal(\"Authorization denied.  The claim does not have any established relationships with the requested resource.\");",
+											"        pm.expect(responseItems.message).to.equal(\"Authorization denied. No relationships have been established between the caller's education organization id claim (value [5001] of type 'CommunityProviderId') and the requested resource's 'SchoolId' value. (Authorization Strategy: RelationshipsWithEdOrgsAndPeopleAuthorizationStrategy`1)\");",
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json]: fixed it

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json [hunk 12]
```diff
@@ -1600,7 +1600,7 @@
 											"});",
 											"const responseItems = pm.response.json();",
 											"pm.test(\"Should match with Error Message Response \", () => {    ",
-											"        pm.expect(responseItems.message).to.equal(\"Authorization denied.  The claim does not have any established relationships with the requested resource.\");",
+											"        pm.expect(responseItems.message).to.equal(\"Authorization denied. No relationships have been established between the caller's education organization id claim (value [255901001] of type 'SchoolId') and the requested resource's 'CommunityProviderId' value. (Authorization Strategy: RelationshipsWithEdOrgsAndPeopleAuthorizationStrategy`1)\");",
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Postman Test Suite/Ed-Fi ODS-API Integration Test Suite Multiple Edorg types.postman_collection.json]: fixed it

File: Postman Test Suite/Ed-Fi ODS-API Integration Test Suite.postman_collection.json [hunk 13]
```diff
@@ -416,7 +416,7 @@
 									"pm.test(\"Should return the item with Invalid SchoolId Validation string\", () => {",
 									"     ",
 									"     const responseItem = pm.response.json();",
-									"     pm.expect(responseItem.message).to.include(\"Authorization denied.  The claim does not have any established relationships with the requested resource\");",
+									"     pm.expect(responseItem.message).to.include(\"Authorization denied. No relationships have been established between the caller's education organization id claim (value [255901] of type 'LocalEducationAgencyId') and the requested resource's 'SchoolId' value. (Authorization Strategy: RelationshipsWithEdOrgsAndPeopleAuthorizationStrategy`1)\");",
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Postman Test Suite/Ed-Fi ODS-API Integration Test Suite.postman_collection.json]: fixed it

File: Application/EdFi.Ods.Common/Security/Authorization/AuthorizationFilterDetails.cs [hunk 14]
```diff
@@ -31,5 +31,7 @@ public class AuthorizationFilterDetails
         /// The values associated with the API client's claim.
         /// </summary>
         public object[] ClaimValues { get; set; }
+
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply Application/EdFi.Ods.Common/Security/Authorization/AuthorizationFilterDetails.cs]: fixed it

**Review by gmcelhanon — APPROVED** [review]

**Pull request merged**